src/main/java/com/hubspot/jinjava/tree/output/OutputList.java

```diff
@@ -3,12 +3,25 @@
 import java.util.LinkedList;
 import java.util.List;
 
+import com.hubspot.jinjava.interpret.OutputTooBigException;
+
 public class OutputList {
 
   private final List<OutputNode> nodes = new LinkedList<>();
   private final List<BlockPlaceholderOutputNode> blocks = new LinkedList<>();
+  private long maxOutputSize;
+
+  public OutputList(long maxOutputSize) {
+
+    this.maxOutputSize = maxOutputSize;
+  }
 
   public void addNode(OutputNode node) {
+
+    if (maxOutputSize > 0 && node.getSize() > maxOutputSize) {
```

Comment:
Does this only check the size of this node against the max? What if we're adding a large number of very small nodes? Just pushed a test that shows what I mean.

If this is only intended to limit the size of individual nodes, I think we should change the name of `maxOutputSize` to something else.